commit e00ee6d530d5f5c656f75270892d94f507feb5c5
Author: Todd Lipcon <todd@cloudera.com>
Date:   Fri May 22 12:14:19 2009 -0700

    Hold off JT RPC startup until after JT is ready to service heartbeats
    
    Fixes a regression in the HADOOP-3746 backport to 0.18.3

diff --git src/mapred/org/apache/hadoop/mapred/JobTracker.java src/mapred/org/apache/hadoop/mapred/JobTracker.java
index af318e7..adc8b6f 100644
--- src/mapred/org/apache/hadoop/mapred/JobTracker.java
+++ src/mapred/org/apache/hadoop/mapred/JobTracker.java
@@ -134,6 +134,8 @@ public class JobTracker implements MRConstants, InterTrackerProtocol,
       try {
         result = new JobTracker(conf);
         result.taskScheduler.setTaskTrackerManager(result);
+        result.interTrackerServer.start();
+
         break;
       } catch (VersionMismatch e) {
         throw e;
@@ -629,7 +631,6 @@ public class JobTracker implements MRConstants, InterTrackerProtocol,
             DNSToSwitchMapping.class), conf);
 
     this.interTrackerServer = RPC.getServer(this, addr.getHostName(), addr.getPort(), handlerCount, false, conf);
-    this.interTrackerServer.start();
     if (LOG.isDebugEnabled()) {
       Properties p = System.getProperties();
       for (Iterator it = p.keySet().iterator(); it.hasNext();) {
